pages/page.py

```diff
@@ -5,6 +5,34 @@
 
 class Page(object):
 
-    def __init__(self, testsetup):
-        self.testsetup = testsetup
-        self.selenium = testsetup.selenium
+    _url = None
+
+    def __init__(self, base_url, selenium):
+        self.base_url = base_url
+        self.selenium = selenium
+
+    def open(self):
+        self.selenium.get(self.url)
+        self.wait_for_page_to_load()
+        return self
+
+    @property
+    def url(self):
+        if self._url is not None:
+            return self._url.format(base_url=self.base_url)
+        return self.base_url
+
+    def wait_for_page_to_load(self):
+        return self
```

Comment:
I started this in Bedrock, because `open` returns the page object, but you might also instantiate a page object without explicitly opening it. It means you can `return NewPage(base_url, selenium).wait_for_page_to_load()` inside a method that clicks a link (for example).